aws-athena/views/reporting-vw_assessment_roll_muni.sql

```diff
@@ -0,0 +1,114 @@
+-- Gathers AVs by year, major class, assessment stage, and municipality for
+-- reporting
+
+/* Ensure every municipality/class/year has a row for every stage through
+cross-joining. This is to make sure that combinations that do not yet
+exist in iasworld.asmt_all for the current year will exist in the view, but have
+largely empty columns. For example: even if no class 4s in the City of Chicago
+have been mailed yet for the current assessment year, we would still like an
+empty City of Chicago/class 4 row to exist for the mailed stage. */
+WITH stages AS (
+
+    SELECT 'mailed' AS stage
+    UNION
+    SELECT 'assessor certified' AS stage
+    UNION
+    SELECT 'bor certified' AS stage
+
+),
+
+/* This CTE removes historical PINs in reporting.vw_pin_township_class that are
+not in reporting.vw_pin_value_long. We do this to make sure the samples we
+derive the numerator and denominator from for stage_portion are the same. These
+differences are inherent in the tables that feed these views (iasworld.pardat
+and iasworld.asmt_all, respectively) and are data errors - not emblematic of
+what portion of a municipality has actually progressed through an assessment
+stage.
+
+It does NOT remove PINs from the most recent year of
+reporting.vw_pin_township_class since we expect these differences based on how
+iasworld.asmt_all is populated through the year as the assessment cycle
+progresses.
```

Comment:
**question:** Won't this not quite work when we're just into a new year i.e. if we roll over into 2025 with a new `taxyr` in `pardat` (which feeds `vw_pin_township_class`), there will still be stage records missing from the Board in `vw_pin_value_long`?

Reply:
I think we'll need to discuss this in person. This part is complicated, but we're already in a situation where we don't have a complete set of PINs in `vw_pin_value_long` for 2024 and the view is working as intended.